From f12d07e0d27cf75f36131fa98e304559058f13ac Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Tue, 11 Oct 2016 17:37:50 +1100
Subject: [PATCH 22/80] Wrong again

---
 kernel/sched/MuQSS.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 3b72a84..d49bca2 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -1859,6 +1859,7 @@ try_to_wake_up(struct task_struct *p, unsigned int state, int wake_flags)
 	smp_cond_load_acquire(&p->on_cpu, !VAL);
 
 	p->sched_contributes_to_load = !!task_contributes_to_load(p);
+	p->state = TASK_WAKING;
 
 	cpu = select_best_cpu(p);
 	if (task_cpu(p) != cpu)
-- 
2.7.4

