From 2c80363bf65443288424b765cf747ce8ebf91fed Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Thu, 13 Oct 2016 11:28:27 +1100
Subject: [PATCH 30/80] Bump MuQSS version to 0.111

---
 kernel/sched/MuQSS.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 2306ab78..58d8c54 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -139,7 +139,7 @@
 
 void print_scheduler_version(void)
 {
-	printk(KERN_INFO "MuQSS CPU scheduler v0.110 by Con Kolivas.\n");
+	printk(KERN_INFO "MuQSS CPU scheduler v0.111 by Con Kolivas.\n");
 }
 
 /*
-- 
2.7.4

