From 7e45f4ac7800785cd1f42ea1ace95fce3e96936f Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 15 Oct 2016 13:48:30 +1100
Subject: [PATCH 40/80] Time slice expire should be set to correct rq

---
 kernel/sched/MuQSS.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 405a5c0..19ac2f7 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -2214,7 +2214,7 @@ void wake_up_new_task(struct task_struct *p)
 			 */
 			rq_curr->time_slice = 0;
 			__set_tsk_resched(rq_curr);
-			time_slice_expired(p, rq);
+			time_slice_expired(p, new_rq);
 			if (suitable_idle_cpus(p))
 				resched_best_idle(p, task_cpu(p));
 			else if (unlikely(rq != new_rq))
@@ -2232,7 +2232,7 @@ void wake_up_new_task(struct task_struct *p)
 				try_preempt(p, new_rq);
 		}
 	} else {
-		time_slice_expired(p, rq);
+		time_slice_expired(p, new_rq);
 		try_preempt(p, new_rq);
 	}
 	double_rq_unlock(rq, new_rq);
-- 
2.7.4

