From ab70a89ec65270c99621839ada1f026065fadca8 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 15 Oct 2016 13:54:04 +1100
Subject: [PATCH 41/80] Remove redundant reassignment

---
 kernel/sched/MuQSS.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 19ac2f7..3a89dc7 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -3357,8 +3357,6 @@ static void task_running_tick(struct rq *rq)
 	} else if (p->time_slice >= RESCHED_US)
 			return;
 out_resched:
-	p = rq->curr;
-
 	rq_lock(rq);
 	__set_tsk_resched(p);
 	rq_unlock(rq);
-- 
2.7.4

