From 755fcbd33b7c03d1c7268a63dcf29d94b6a05b49 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Mon, 17 Oct 2016 08:33:10 +1100
Subject: [PATCH 45/80] Don't re-set values that haven't changed.

---
 kernel/sched/MuQSS.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 39cc843..7d48f69 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -3849,10 +3849,7 @@ static void __sched notrace __schedule(bool preempt)
 	clear_preempt_need_resched();
 
 	if (idle != prev) {
-		/* Update all the information stored on struct rq */
-		prev->deadline = rq->rq_deadline;
 		check_deadline(prev, rq);
-		prev->last_ran = rq->clock_task;
 		return_task(prev, rq, cpu, deactivate);
 	}
 
-- 
2.7.4

