From 2ddc12e75f51ad54a8cc814bc7e5f3ec184de203 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Mon, 17 Oct 2016 14:48:53 +1100
Subject: [PATCH 50/80] Bump MuQSS version to 0.112

---
 kernel/sched/MuQSS.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 02644e2..1700be5 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -134,7 +134,7 @@
 
 void print_scheduler_version(void)
 {
-	printk(KERN_INFO "MuQSS CPU scheduler v0.111 by Con Kolivas.\n");
+	printk(KERN_INFO "MuQSS CPU scheduler v0.112 by Con Kolivas.\n");
 }
 
 /*
-- 
2.7.4

