From 6caa689a9a9d2817e2a77a033dc32b37e3cd3d68 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Fri, 21 Oct 2016 08:04:32 +1100
Subject: [PATCH 70/80] Clamp niffy diff to max diff if it's over instead of
 min diff.

---
 kernel/sched/MuQSS.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 513efb1..f6df670 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -280,8 +280,10 @@ static inline void niffy_diff(s64 *niff_diff, int jiff_diff)
 	/*  Round up to the nearest tick for maximum */
 	max_diff = JIFFIES_TO_NS(jiff_diff + 1);
 
-	if (unlikely(*niff_diff < min_diff || *niff_diff > max_diff))
+	if (unlikely(*niff_diff < min_diff))
 		*niff_diff = min_diff;
+	else if (unlikely(*niff_diff > max_diff))
+		*niff_diff = max_diff;
 }
 
 #ifdef CONFIG_SMP
-- 
2.7.4

